sanitycheck: handle timeouts correctly with qemu

nashif opened the pull request:
In some cases we were not handling return code correctly and evaluating
timeouts as a pass. Report failure correctly.

Fixes #26065

tyrantlizard77 reviewed (approved):
This patch works for me, I noted this issue when testing #26229

wentongwu commented on scripts/sanity_chk/sanitylib.py:
```diff
@@ -902,16 +905,18 @@ def handle(self):
-            else:
-                self.returncode = proc.returncode
 
+            self.returncode = proc.returncode
```
if put `self.returncode = proc.returncode` here, the returncode in case
```
                if os.path.exists(self.pid_fn):
                    qemu_pid = int(open(self.pid_fn).read())
                    try:
                        os.kill(qemu_pid, signal.SIGKILL)
                    except ProcessLookupError:
                        pass
                    proc.wait()
                    self.returncode = 0
```
will be overwrite.

nashif replied:
isnt this the point? Why are you passing 0 return code in case of a timeout?

wentongwu replied:
there is a comment about this case, it intents to handle QEMU doesn't handle SIGTERM signal timely and correctly in sanitycheck environment(for example sample/Philosopher case), send SIGKILL signal to QEMU process and let sanitycheck judge the result by console output. 
```
#sometimes QEMU can't handle SIGTERM signal correctly
#in that case kill -9 QEMU process directly and leave
#sanitycheck judge testing result by console output
```

nashif replied:
can you suggest a fix then, since you were the one who introduced this bug?

wentongwu replied:
will do.

wentongwu replied:
the fix will be like below:
```
diff --git a/scripts/sanity_chk/sanitylib.py b/scripts/sanity_chk/sanitylib.py
index a083c34b64..b2fa4d11f2 100644
--- a/scripts/sanity_chk/sanitylib.py
+++ b/scripts/sanity_chk/sanitylib.py
@@ -898,7 +898,10 @@ class QEMUHandler(Handler):
                     except ProcessLookupError:
                         pass
                     proc.wait()
-                    self.returncode = 0
+                    if harness.state == "passed":
+                        self.returncode = 0
+                    else:
+                        self.returncode = proc.returncode
                 else:
                     proc.terminate()
                     proc.kill()
```

PerMac commented:
I've updated the #26187 PR and move out from draft. I think it is better not to include cb080b1 commit here as it will be handled by #26187 and separation of issues will be cleaner.

wentongwu committed:
sanitycheck: handle timeouts correctly with qemu

In some cases we were not handling return code correctly and evaluating
timeouts as a pass. Report failure correctly.

Fixes #26065

Signed-off-by: Wentong Wu <wentong.wu@intel.com>

nashif committed:
tests: jwt: increase timeout for test

Some platforms need some more time here.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
boards: qemu_cortex_a53: do not run on net/bt tests

Those are covered by other qemu variants..

Signed-off-by: Anas Nashif <anas.nashif@intel.com>
